### PR TITLE
feat(konnect): require gw discovery, rename gw discovery and release 2.17.0-rc.1

### PR DESCRIPTION
**Pull request by czeslavo**
#### What this PR does / why we need it:

Requires Gateway discovery to be enabled when Konnect sync is enabled. As a result, splits the example values for Konnect into the already existing `minimal-kong-gd-gateway.yaml` and a new `minimal-kong-gd-controller-konnect.yaml`. Also renames the `serviceDiscovery` to `gatewayDiscovery`.

#### Which issue this PR fixes

Part of https://github.com/Kong/kubernetes-ingress-controller/issues/3581 and #740.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] PR is based off the current tip of the `main` branch.
- [x] Changes are documented under the "Unreleased" header in CHANGELOG.md
- [x] New or modified sections of values.yaml are documented in the README.md
- [x] Commits follow the [Kong commit message guidelines](https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#commit-message-format)

### CODE REVIEW

File: charts/kong/README.md [hunk 1]
```diff
@@ -701,39 +701,39 @@ are configured using an array of objects under `proxy.stream` and `udpProxy.stre
...
+| konnect.enabled                            | Enable synchronisation of data plane configuration with Konnect Runtime Group                                                                            | false                              |
+| konnect.runtimeGroupID                     | Konnect Runtime Group's unique identifier.                                                                                                               |                                    |
+| konnect.region                             | Konnect account's region. One of `us`, `eu`.                                                                                                             | us                                 |
+| konnect.tlsClientCertSecretName            | Name of the secret that contains Konnect Runtime Group's client TLS certificate.                                                                         | konnect-client-tls                 |
+
+[sd_section]: #the-gatewayDiscovery-section
```
Comment (mflendrich) [inline charts/kong/README.md]: nit: `sd` -> `gd`

**Review by mflendrich — DISMISSED** [review]
Let's go.

In search of renaming leftovers I have: eyeballed the diff, grepped the codebase for `sd`, `service.discovery` and `servicediscovery` - all in a case-insensitive way. Looks good.

File: charts/kong/README.md (hunk 1 above)
Comment (czeslavo) [reply charts/kong/README.md]: @mflendrich 👍 Fixed, PTAL

Comment (mflendrich) [reply charts/kong/README.md]: argh, force pushed :)
luckily I had the old version in the local working copy and could `git diff`. Fix LGTM

**Review by mflendrich — APPROVED** [review]

**Pull request merged**